streaming/vision/base.py

```diff
@@ -45,33 +65,46 @@ def __init__(self,
         super().__init__(local, remote, split, shuffle, prefetch, keep_zip, retry, timeout, hash,
                          batch_size)
 
+        if split not in ['train', 'val']:
+            raise ValueError(f"split='{split}' must be one of ['train', 'val'].")
+
         has_transforms = transforms is not None
         has_separate_transform = transform is not None or target_transform is not None
         if has_transforms and has_separate_transform:
-            raise ValueError('Only transforms or transform/target_transform can be passed as ' +
-                             'argument')
+            raise ValueError(
+                'Only transforms or transform/target_transform can be passed as an argument')
 
         self.transform = transform
         self.target_transform = target_transform
-
-        if not has_transforms:
-            transforms = StandardTransform(transform, target_transform)
         self.transforms = transforms
 
-    def __getitem__(self, idx: int) -> Any:
-        obj = super().__getitem__(idx)
-        x = obj['x']
-        y = obj['y']
-        return self.transforms(x, y)
-
 
 class ImageClassDataset(VisionDataset):
+    """
+    Base Class for creating an Image Classification streaming datasets. It is recommend to override the ``__getitem__`` method.
+
+    Args:
+        local (str): Local filesystem directory where dataset is cached during operation.
+        split (str): The dataset split to use, either 'train' or 'val'.
+        remote (str, optional): Remote directory (S3 or local filesystem) where dataset is stored. Default: ``None``.
+        shuffle (bool, optional): Whether to shuffle the train samples in this dataset. Default: ``True``.
+        transforms (callable, optional): A function/transforms that takes in an image and a label and returns the transformed versions of both. Default: ``None``.
```

Comment:
< 100 col

Reply:
I rearranged it for now manually but we need some automation to check the doc max length. I will look into it in next PR.